core/src/main/java/org/apache/accumulo/core/file/rfile/RelativeKey.java

```diff
@@ -432,24 +432,24 @@ public static SkippR fastSkip(DataInput in, Key seekKey, MutableByteSequence val
     return new SkippR(result, count, newPrevKey);
   }
 
-  private static void read(DataInput in, MutableByteSequence mbseq) throws IOException {
+  private static void read(DataInput in, ArrayByteSequence mbseq) throws IOException {
     int len = WritableUtils.readVInt(in);
     read(in, mbseq, len);
   }
 
-  private static void readValue(DataInput in, MutableByteSequence mbseq) throws IOException {
+  private static void readValue(DataInput in, ArrayByteSequence mbseq) throws IOException {
     int len = in.readInt();
     read(in, mbseq, len);
   }
 
-  private static void read(DataInput in, MutableByteSequence mbseqDestination, int len)
+  private static void read(DataInput in, ArrayByteSequence mbseqDestination, int len)
       throws IOException {
     if (mbseqDestination.getBackingArray().length < len) {
-      mbseqDestination.setArray(new byte[UnsynchronizedBuffer.nextArraySize(len)], 0, 0);
+      mbseqDestination.reset(new byte[UnsynchronizedBuffer.nextArraySize(len)], 0, 0);
     }
 
     in.readFully(mbseqDestination.getBackingArray(), 0, len);
-    mbseqDestination.setLength(len);
+    mbseqDestination.reset(mbseqDestination.getBackingArray(), mbseqDestination.offset(), len);
```

Comment:
I think this could be redone as:

```
    if (mbseqDestination.getBackingArray().length < len) {
      byte[] buf = new byte[UnsynchronizedBuffer.nextArraySize(len);
      in.readFully(buf, 0, len);
      mbseqDestination.reset(buf, 0, len);
    } else {
      in.readFully(mbseqDestination.getBackingArray(), 0, len);
      mbseqDestination.reset(mbseqDestination.getBackingArray(), mbseqDestination.offset(), len);
    }
```

Reply:
> I think this could be redone as:
> 
> ```
>     if (mbseqDestination.getBackingArray().length < len) {
>       byte[] buf = new byte[UnsynchronizedBuffer.nextArraySize(len);
>       in.readFully(buf, 0, len);
>       mbseqDestination.reset(buf, 0, len);
>     } else {
>       in.readFully(mbseqDestination.getBackingArray(), 0, len);
>       mbseqDestination.reset(mbseqDestination.getBackingArray(), mbseqDestination.offset(), len);
>     }
> ```

Actually, it can be further simplified:

```
    byte[] buf = dest.getBackingArray();
    if (dest.getBackingArray().length < len) {
      buf = new byte[UnsynchronizedBuffer.nextArraySize(len)];
    }
    in.readFully(buf, 0, len);
    mbseqDestination.reset(buf, 0, len);
```

Reply:
Could simplify a bit further.

```java
    byte[] buf = dest.getBackingArray();
    if (buf.length< len) {
      buf = new byte[UnsynchronizedBuffer.nextArraySize(len)];
    }
    in.readFully(buf, 0, len);
    mbseqDestination.reset(buf, 0, len);
```

Reply:
Thanks much, updated.